services/process-service.ts

```diff
@@ -0,0 +1,40 @@
+import * as fiberBootstrap from "../fiber-bootstrap";
+
+export class ProcessService implements IProcessService {
+	private static PROCESS_EXIT_SIGNALS = ["exit", "SIGINT", "SIGTERM"];
+	private _listeners: IListener[];
+
+	public get listenersCount(): number {
+		return this._listeners.length;
+	}
+
+	constructor() {
+		this._listeners = [];
+		_.each(ProcessService.PROCESS_EXIT_SIGNALS, (signal: string) => {
```

Comment:
this could be simplified to:
```TypeScript
_.each(ProcessService.PROCESS_EXIT_SIGNALS, signal => process.on(signal, () => this.executeAllCallbacks.apply(this)));
```
but I do not insist on changing it as it may be harder to read.